### PR TITLE
Update .htaccess

### PR DESCRIPTION
**Pull request by Luceos**
flarum/core#692 protect composer.lock, but also improve protection for git directories

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -26,7 +26,8 @@
   RewriteRule ^vendor/(.*)?$ / [F,L]
   RewriteRule ^storage/(.*)?$ / [F,L]
   RewriteRule ^config.php$ / [F,L]
-  RewriteRule ^.git/(.*)?$ / [F,L]
+  RewriteRule /\.git / [F,L]
```
Comment (franzliedke) [inline .htaccess]: With that leading slash, does this still take care of `.git` in the root directory?

File: .htaccess [hunk 2]
```diff
@@ -26,7 +26,8 @@
...
+  RewriteRule ^composer\.lock$ / [F,L]
```
Comment (franzliedke) [inline .htaccess]: What about something like `^composer\.(.+)$` to match composer.json as well?

**Comment by franzliedke** [comment]
And we need to do the same for the Nginx config in our docs...

File: .htaccess (hunk 2 above)
Comment (Luceos) [inline .htaccess]: @franzliedke I thought about it, but that wasn't in the scope of the referenced issue 😁

Comment (franzliedke) [inline .htaccess]: Honestly, I think it should be done together. That's pretty much our only hope of keeping these things in sync. ;)

File: .htaccess (hunk 1 above)
Comment (Luceos) [inline .htaccess]: It's taken from the [example](http://stackoverflow.com/questions/6142437/make-git-directory-web-inaccessible/17916515#17916515). The leading slash matched the root of the rewriting base, I've double checked it in the apache logs.

File: .htaccess (hunk 2 above)
Comment (Luceos) [inline .htaccess]: Do you agree `^composer\.(json|lock)$ suffices?

**Comment by Luceos** [comment]
@arkinn are you able to create a docs issue and/or add the required update? Feel free to refer/mention back.

**Comment by Luceos** [comment]
@franzliedke can we merge this and create an issue for Nginx, I'll gladly create that once merged.

**Comment by tobscure** [comment]
@luceos done

**Pull request merged**